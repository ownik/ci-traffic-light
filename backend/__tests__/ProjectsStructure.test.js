const ProjectsStructure = require('../src/ProjectsStructure');

describe('ProjectsStructure', () => {
  const structure = new ProjectsStructure()
    .addProject('Windows', 'Windows Project Name', null)
    .addProject('WindowsSubProject1', 'Windows SubProject Name', 'Windows')
    .addBuild('WindowsSubProject1Build1', 'Build 1', 'WindowsSubProject1')
    .addBuild('WindowsSubProject1Build2', 'Build 2', 'WindowsSubProject1')
    .addBuild('WindowsSubProject1Build3', 'Build 3', 'WindowsSubProject1')
    .addProject('Linux', 'Linux Project Name', null)
    .addBuild('LinuxBuild1', 'Build 1', 'Linux')
    .addBuild('LinuxBuild2', 'Build 2', 'Linux')
    .addBuild('LinuxBuild3', 'Build 3', 'Linux');

  console.log(JSON.stringify(structure._projects));

  test.each([
    { id: 'Windows', expectedName: 'Windows Project Name' },
    { id: 'WindowsSubProject1Build3', expectedName: 'Build 3' },
    { id: 'NotExistedId', expectedName: 'NotExistedId' },
  ])('$expected when state is $state', async ({ id, expectedName }) => {
    expect(structure.getName(id)).toEqual(expectedName);
  });

  test('toString', () => {
    expect(structure.toString()).toMatchSnapshot();
  });
});
