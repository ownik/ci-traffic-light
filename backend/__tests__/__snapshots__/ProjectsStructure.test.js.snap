// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectsStructure toString 1`] = `
"null - null
  Windows - Windows Project Name
    WindowsSubProject1 - Windows SubProject Name
      WindowsSubProject1Build1 - Build 1
      WindowsSubProject1Build2 - Build 2
      WindowsSubProject1Build3 - Build 3
  Linux - Linux Project Name
    LinuxBuild1 - Build 1
    LinuxBuild2 - Build 2
    LinuxBuild3 - Build 3
"
`;
